### PR TITLE
update: updated notation verify command based on spec

### PR DESCRIPTION
**Pull request by patrickzheng200**
This PR updates notation verify command based on the [verify_spec](https://github.com/notaryproject/notation/blob/main/specs/commandline/verify.md).

Tested with `notation verify` on ACR.

Signed-off-by: Patrick Zheng <patrickzheng@microsoft.com>

### CODE REVIEW

**Comment by JeyJeyGao** [comment]
LGTM

**Review by JeyJeyGao — APPROVED** [review]
LGTM

File: cmd/notation/verify.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ notation verify [--plugin-config <key>=<value>,...] [--username <username>] [--p
 		},
 	}
 	opts.ApplyFlags(command.Flags())
-	command.Flags().StringVarP(&opts.pluginConfig, "plugin-config", "c", "", "list of comma-separated {key}={value} pairs that are passed as is to the plugin")
+	command.Flags().StringVarP(&opts.pluginConfig, "plugin-config", "c", "", "{key}={value} pairs that are passed as is to a plugin, if the verification is associated with a verification plugin, refer plugin documentation to set appropriate values")
```
Comment (shizhMSFT) [inline cmd/notation/verify.go:47]: `opts.pluginConfig` should be `[]string` instead of `string`.

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

**Review by jondonas — APPROVED** [review]
LGTM

File: cmd/notation/verify.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ notation verify [--plugin-config <key>=<value>,...] [--username <username>] [--p
 		},
 	}
 	opts.ApplyFlags(command.Flags())
-	command.Flags().StringVarP(&opts.pluginConfig, "plugin-config", "c", "", "list of comma-separated {key}={value} pairs that are passed as is to the plugin")
+	command.Flags().StringArrayVarP(&opts.pluginConfig, "plugin-config", "c", []string{}, "{key}={value} pairs that are passed as it is to a plugin, if the verification is associated with a verification plugin, refer plugin documentation to set appropriate values")
```
Comment (shizhMSFT) [inline cmd/notation/verify.go:47]: nit:
```suggestion
	command.Flags().StringArrayVarP(&opts.pluginConfig, "plugin-config", "c", nil, "{key}={value} pairs that are passed as it is to a plugin, if the verification is associated with a verification plugin, refer plugin documentation to set appropriate values")
```

File: internal/cmd/flags.go [hunk 3]
```diff
@@ -73,12 +72,12 @@ var (
...
-		Name:      "pluginConfig",
+		Name:      "plugin-config",
 		Shorthand: "c",
-		Usage:     "list of comma-separated {key}={value} pairs that are passed as is to the plugin, refer plugin documentation to set appropriate values",
+		Usage:     "{key}={value} pairs that are passed as it is to a plugin, refer plugin's documentation to set appropriate values",
 	}
-	SetPflagPluginConfig = func(fs *pflag.FlagSet, p *string) {
-		fs.StringVarP(p, PflagPluginConfig.Name, PflagPluginConfig.Shorthand, "", PflagPluginConfig.Usage)
+	SetPflagPluginConfig = func(fs *pflag.FlagSet, p *[]string) {
+		fs.StringArrayVarP(p, PflagPluginConfig.Name, PflagPluginConfig.Shorthand, []string{}, PflagPluginConfig.Usage)
```
Comment (shizhMSFT) [inline internal/cmd/flags.go:80]: nit:
```suggestion
		fs.StringArrayVarP(p, PflagPluginConfig.Name, PflagPluginConfig.Shorthand, nil, PflagPluginConfig.Usage)
```

File: internal/cmd/flags.go [hunk 4]
```diff
@@ -88,94 +87,14 @@ type KeyValueSlice interface {
...
-			if _, exist := m[k]; exist {
-				return nil, fmt.Errorf("duplicated key: %q", k)
-			}
-			m[k] = v
-		} else {
-			return nil, fmt.Errorf("malformed entry: %q", c)
+func ParseFlagPluginConfig(config []string) (map[string]string, error) {
+	pluginConfig := make(map[string]string, len(config))
+	for _, pair := range config {
+		index := strings.Index(pair, "=")
```
Comment (shizhMSFT) [inline internal/cmd/flags.go:93]: nit: Try [strings.Cut()](https://pkg.go.dev/strings#Cut).

**Review by shizhMSFT — APPROVED** [review]
LGTM with suggestions

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**